codex/stores/memorystore.nim

```diff
@@ -0,0 +1,182 @@
+## Nim-Codex
+## Copyright (c) 2021 Status Research & Development GmbH
+## Licensed under either of
+##  * Apache License, version 2.0, ([LICENSE-APACHE](LICENSE-APACHE))
+##  * MIT license ([LICENSE-MIT](LICENSE-MIT))
+## at your option.
+## This file may not be copied, modified, or distributed except according to
+## those terms.
+
+import pkg/upraises
+
+push: {.upraises: [].}
+
+import std/options
+import std/tables
+import std/lists
+
+import pkg/chronicles
+import pkg/chronos
+import pkg/libp2p
+import pkg/questionable
+import pkg/questionable/results
+
+import ./blockstore
+import ./consts
+import ../chunker
+import ../manifest
+
+export blockstore
+
+logScope:
+  topics = "codex memorystore"
+
+type
+  MemoryStoreNode = ref object
+    key: Cid
+    val: Block
+
+  MemoryStore* = ref object of BlockStore
```

Comment:
In general, I think the `CacheStore` already did this, what is the reason to re-implement it?

Reply:
Furthermore, I think if we're going for something more generic, I would implement this as a datastore and use it with the tiered datastore abstraction to enable caching at any layer.